Dockerfile

```diff
@@ -2,20 +2,18 @@
 # supports Debian 11
 FROM python:3.8-slim-buster AS build-image
 
-RUN apt update
-RUN apt full-upgrade -y
-RUN apt install -y --no-install-recommends \
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
     build-essential \
     curl \
     default-libmysqlclient-dev \
     libpq-dev \
     unixodbc-dev \
     unzip
 
-ADD . /srcdir
+COPY . /srcdir
 RUN python3 -m venv /virtualenv
 ENV PATH="/virtualenv/bin:$PATH"
-RUN pip install --upgrade pip
```

Comment:
I don't see a reason to remove this, better to always upgrade to the latest `pip`

Reply:
For the same reason it's suggested to stick to an exact version in `requirements.txt`, upgrading to a newer version of any package in Dockerfile would lead to a different docker image for different users running it in different times which would make debugging harder and may break the docker image for some users which you cannot catch these unexpected errors like [this](https://github.com/pypa/pip/issues/10219).